### PR TITLE
NOTICK Extend test app and smoke tests for UTXO ledger

### PR DESCRIPTION
**Pull request by blsemo**
Extend the UTXO demo app to incude:
- evolving a state (consuming states)
- loading inputs and outputs from a LedgerTransaction

Also extend the UTXO ledger smoketest to exercise all that logic.

### CODE REVIEW

File: testing/cpbs/ledger-utxo-demo-app/src/test/kotlin/net/cordapp/demo/utxo/TestPeekTransactionFlow.kt [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+            flow.marshallingService = jsonMarshallingService
+            flow.ledgerService = ledgerService
+
+            val goodRequest = mock<RPCRequestData>()
+            val body = PeekTransactionParameters(txIdGood.toString())
+            whenever(goodRequest.getRequestBodyAs<PeekTransactionParameters>(jsonMarshallingService)).thenReturn(body)
+
+            val result = flow.call(goodRequest)
+            println(result)
```
Comment (vlajos) [inline testing/cpbs/ledger-utxo-demo-app/src/test/kotlin/net/cordapp/demo/utxo/TestPeekTransactionFlow.kt]: I this a debug left over?

File: testing/cpbs/ledger-utxo-demo-app/src/main/kotlin/net/cordapp/demo/utxo/PeekTransactionFlow.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+        val txId = requestObject.transactionId
+
+        log.info("Utxo finding transaction $txId")
+        val ledgerTransaction = ledgerService.findLedgerTransaction(SecureHash.parse(txId))
+
+
+        val resultString = marshallingService.format(extractStates(ledgerTransaction))
+
+        log.info("Utxo finding transaction $txId's result: $resultString")
```
Comment (vlajos) [inline testing/cpbs/ledger-utxo-demo-app/src/main/kotlin/net/cordapp/demo/utxo/PeekTransactionFlow.kt]:
```suggestion
        log.info("Utxo peeking transaction $txId's result: $resultString")
```

**Review by vlajos — COMMENTED** [review]
I think the e2e repository would be a better place for this.
Otherwise 2 minor comments.

**Review by syamkovar3 — COMMENTED** [review]
final Rc for beta1 was done today  , see https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4348018689/C5+Beta+Program+-+Fox+1.1+Release. should this pr be merged to main 5.0 ?

**Comment by blsemo** [comment]
> I think the e2e repository would be a better place for this. Otherwise 2 minor comments.

I disagree - I think this is extremely valuable as a smoke test in the PR gate - it just extends an existing test, so doesn't add a lot of runtime overhead, but significantly increases coverage.

**Review by vlajos — APPROVED** [review]
LGTM

**Pull request merged**